src/components/Charts/TradingViewAreaChart.tsx

```diff
@@ -0,0 +1,152 @@
+import { createChart, UTCTimestamp } from 'lightweight-charts'
+import { useEffect, useRef, useState } from 'react'
+import { Box, Flex } from 'rebass'
+
+import { TYPE } from '../../theme/index'
+import { PricePercentualDifference } from './PricePercentualDifference'
+import {
+  buildDate,
+  ChartData,
+  formatDate,
+  formatDateHours,
+  formatDateShort,
+  formatPrice,
+  lastDataElement,
+  lastElementTimeOrDefault,
+  lastElementValueOrDefault,
+} from './simpleChartUtils'
+
+const chartColors = {
+  backgroundColor: 'transparent',
+  textColor: '#8780BF',
+  lineColor: '#c8bdff',
+  areaTopColor: 'rgba(255, 255, 255, 0.6)',
+  areaBottomColor: 'rgba(204, 144, 255, 0)',
+}
+
+type TradingViewAreaChartProps = {
+  data: ChartData[]
+  tokenSymbol?: string
+  showHours?: boolean
+}
+
+export const TradingViewAreaChart = ({ data, tokenSymbol, showHours }: TradingViewAreaChartProps) => {
+  const chartRef = useRef<HTMLDivElement>(null)
+  const [price, setPrice] = useState(lastElementValueOrDefault(data))
+  const [date, setDate] = useState(buildDate(lastElementTimeOrDefault(data)))
+
+  useEffect(() => {
+    if (
+      !chartRef ||
```

Comment:
maybe we can move it to a separate hook file, it will be a bit cleaner

Reply:
hmm.. let me think about it.

Reply:
I tried it and I think it's better with the hook. Will update the branch shortly. Thanks 🙌

Reply:
Done ✔️